spec/ddtrace/contrib/httparty/example_client_spec.rb

```diff
@@ -0,0 +1,82 @@
+require 'spec_helper'
+require 'ddtrace'
+require 'httparty'
+
+RSpec.describe 'ExampleClient' do
+  let(:tracer) { get_test_tracer }
+  let(:configuration_options) { { tracer: tracer } }
+
+  before do
+    Datadog.configure do |c|
+      c.use :httparty, configuration_options
+    end
+
+    WebMock.disable_net_connect!
```

Comment:
I'm all for preventing outgoing HTTP requests, but we've had a number of problems with WebMock. Is there anyway to use RSpec stubs/mocks to do this in a way that's not overly burdensome or brittle?

Reply:
This has been very stable in my own testing, and to do something with mocks would require a lot of digging through both the HTTParty gem, and Net::HTTP, which at the end of the day could lead to more brittle tests if implementations change in the future.

I made a few attempts at doing some simple mocking, and one tough issue I ran into was that rspec does not allow mocking of methods that have been prepended.